drivers/rn2xx3/rn2xx3_getset.c

```diff
@@ -334,13 +333,10 @@ void rn2xx3_mac_set_rx2_freq(rn2xx3_t *dev, uint32_t freq)
 {
     dev->loramac.rx2_freq = freq;
     dev->loramac.rx2_dr = rn2xx3_mac_get_rx2_dr(dev);
-    memset(dev->cmd_buf, 0, strlen(dev->cmd_buf));
-    snprintf(dev->cmd_buf, 13, "%s %s %s ",
-             mac, set, "rx2");
 
-    char value_str[16];
-    snprintf(value_str, 16, "%d %lu", dev->loramac.rx2_dr, freq);
-    strcat(dev->cmd_buf, value_str);
+    dev->cmd_buf[0] = 0;
```

Comment:
This cmd_buf[p] = 0 must be after snprintf. Before snprintf it is useless code. (See other remark about this. Please check all other snprintf)